File: src/JsBind.Net/content/JsBindScripts/Modules/AccessPaths.js
const ReferenceIdPrefix = "#";
const AccessPathSeparator = ".";

class AccessPathsClass {
  /**
   * Checks if the is a reference identifier.
   * @param {string} accessPath
   * @returns {boolean}
   */
  isReferenceId(accessPath) {
    return !!accessPath && accessPath.startsWith(ReferenceIdPrefix);
  }

  /**
   * Gets the reference identifier from the access path.
   * @param {string} accessPath
   * @returns {string}
   */
  getReferenceId(accessPath) {
    return accessPath?.substring(ReferenceIdPrefix.length);
  }

  /**
   * Get access path from the specified reference identifier.
   * @param {string} referenceId The object reference identifier.
   * @returns {string} The access path.
   */
  fromReferenceId(referenceId) {
    return ReferenceIdPrefix + referenceId;
  }

  /**
   * Combine multiple access paths.
   * @param {string[]} accessPaths
   * @returns {string}
   */
  combine(...accessPaths) {
    return accessPaths?.filter(accessPath => typeof accessPath === "string" && accessPath.length)?.join(AccessPathSeparator);
  }

  /**
   * Splits the access path based on the access path separator.
   * @param {string} accessPath
   * @returns {string[]}
   */
  split(accessPath) {
    return accessPath?.split(AccessPathSeparator);
  }
}

const AccessPaths = new AccessPathsClass();
export default AccessPaths;

File: src/JsBind.Net/AccessPaths.cs
using System;
using System.Linq;

namespace JsBind.Net;

/// <summary>
/// Performs operations on string instances that contains access path in JavaScript.
/// </summary>
public static class AccessPaths
{
    private const string AccessPathSeparator = ".";
    private const string ReferenceIdPrefix = "#";
    private static readonly int ReferenceIdLength = ReferenceIdPrefix.Length + Guid.Empty.ToString().Length;

    /// <summary>
    /// Checks if the access path is a reference identifier.
    /// </summary>
    /// <param name="accessPath">The access path.</param>
    /// <returns><c>true</c> if the access path is a reference identifier, <c>false</c> otherwise.</returns>
    public static bool IsReferenceId(string? accessPath)
        => accessPath is not null && accessPath.StartsWith(ReferenceIdPrefix) && accessPath.Length == ReferenceIdLength;

    /// <summary>
    /// Gets the reference identifier from the access path.
    /// </summary>
    /// <param name="accessPath">The access path.</param>
    /// <returns>The reference identifier.</returns>
    public static Guid? GetReferenceId(string? accessPath)
    {
        var referenceId = accessPath?[ReferenceIdPrefix.Length..];
        if (referenceId is null)
        {
            return null;
        }

        if (Guid.TryParse(referenceId, out var referenceIdGuid))
        {
            return referenceIdGuid;
        }

        return null;
    }

    /// <summary>
    /// Gets the access path from the reference identifier.
    /// </summary>
    /// <param name="referenceId">The reference identifier.</param>
    /// <returns>The access path.</returns>
    public static string? FromReferenceId(Guid? referenceId)
    {
        if (referenceId is null)
        {
            return null;
        }

        return ReferenceIdPrefix + referenceId.Value.ToString();
    }

    /// <summary>
    /// Gets the access path from the reference identifier.
    /// </summary>
    /// <param name="referenceId">The reference identifier.</param>
    /// <returns>The access path.</returns>
    public static string? FromReferenceId(string? referenceId)
    {
        if (referenceId is null)
        {
            return null;
        }

        if (referenceId.Length != Guid.Empty.ToString().Length)
        {
            throw new InvalidOperationException($"{referenceId} is not a valid reference identifier.");
        }

        return ReferenceIdPrefix + referenceId;
    }

    /// <summary>
    /// Combines multiple access paths.
    /// </summary>
    /// <param name="accessPaths">The access paths.</param>
    /// <returns>The combined access path.</returns>
    public static string? Combine(params string?[] accessPaths)
        => string.Join(AccessPathSeparator, accessPaths.Where(accessPath => !string.IsNullOrEmpty(accessPath)));

    /// <summary>
    /// Splits the access path based on the access path separator.
    /// </summary>
    /// <param name="accessPath">The access path.</param>
    /// <returns>The array of individual access paths.</returns>
    public static string[]? Split(string? accessPath)
        => accessPath?.Split(AccessPathSeparator);
}
